From 9ca7b7f2f1b7f15d8d7568db04294132a1987d70 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 25 May 2022 07:59:24 +0300
Subject: [PATCH] [SW-0] rom: update axprot setting

Axprot settings for jtag and twi interfaces have been updated as per
the updated gaudi3 security specification.

By default jtag and twi interfaces are privileged. The FW sets it to
secured when the respective interfaces are to be disabled.

Change-Id: Id89a735e188314c09bceb45d5ff99b4574a12e56
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 41617aa..00ee80e 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -507,12 +507,12 @@
 
 void security_enable_twi_if(void)
 {
-	security_i2c_axprot_config();
-	security_smi_axprot_config();
 }
 
 void security_disable_twi_if(void)
 {
+	security_i2c_axprot_config();
+	security_smi_axprot_config();
 }
 
 void security_handle_twi(void)
@@ -529,11 +529,6 @@
 	regmap_writel(status, mmD0_PSOC_CS_TRACE_SECURITY_CTRL);
 }
 
-void security_disable_jtag_if(void)
-{
-	security_jtag_dap(SECURITY_JTAG_DAP_DISABLE);
-}
-
 static void security_jtag_axprot_config(void)
 {
 	regmap_writel(AXPROT_UNPRIV_VAL, mmD0_JT_MSTR_IF_AXPROT_LBW_AWPROT_0_OVRD);
@@ -546,10 +541,15 @@
 	regmap_writel(AXPROT_SEC_VAL, mmD0_JT_MSTR_IF_AXPROT_HBW_ARPROT_1_OVRD);
 }
 
+void security_disable_jtag_if(void)
+{
+	security_jtag_dap(SECURITY_JTAG_DAP_DISABLE);
+	security_jtag_axprot_config();
+}
+
 void security_enable_jtag_if(void)
 {
 	security_jtag_dap(SECURITY_JTAG_DAP_ENABLE);
-	security_jtag_axprot_config();
 }
 
 void security_handle_jtag_if(void)
